api/v1beta1/machinedeployment_types.go

```diff
@@ -298,6 +298,33 @@ type MachineDeploymentStatus struct {
 	// Conditions defines current service state of the MachineDeployment.
 	// +optional
 	Conditions Conditions `json:"conditions,omitempty"`
+
+	// v1beta2 groups all the fields that will be added or modified in MachineDeployment's status with the V1Beta2 version.
+	// +optional
+	V1Beta2 *MachineDeploymentV1Beta2Status `json:"v1beta2,omitempty"`
+}
+
+// MachineDeploymentV1Beta2Status groups all the fields that will be added or modified in MachineDeployment with the V1Beta2 version.
+// See https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240916-improve-status-in-CAPI-resources.md#machineset-newconditions for more context.
+type MachineDeploymentV1Beta2Status struct {
+	// conditions represents the observations of a MachineDeployment's current state.
+	// Known condition types are MachinesReady, MachinesUpToDate, ScalingUp, ScalingDown, Remediating, Deleting, Paused.
```

Comment:
orthogonal to the PR This makes me think that although it could be maybe inferred from MachinesUpToDate, there's no immediate way to understand a machineDeployment is going through a rollout and what caused it

Reply:
nvm I see we actually mentioned that explicitly here https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240916-improve-status-in-CAPI-resources.md#example-use-cases
I think we'll find value on solving this eventually

Reply:
fixed
Agreed that we should pay attention how we aggregate data at MD level.